src/components/timecard/scheduled-rest-day/ScheduledRestDay.jsx

```diff
@@ -0,0 +1,127 @@
+import { PropTypes } from 'prop-types';
+import { Link } from 'react-router-dom';
+import dayjs from 'dayjs';
+
+import {
+  formatDateTimeISO,
+  midnight,
+  removeTimecardContextEntry,
+} from '../../../utils/time-entry-utils/timeEntryUtils';
+import { UrlSearchParamBuilder } from '../../../utils/api-utils/UrlSearchParamBuilder';
+import { useTimecardContext } from '../../../context/TimecardContext';
+import { deepClone } from '../../../utils/common-utils/common-utils';
+import {
+  createTimeEntry,
+  deleteTimeEntry,
+} from '../../../api/services/timecardService';
+import { ContextTimeEntry } from '../../../utils/time-entry-utils/ContextTimeEntry';
+
+const ScheduledRestDay = ({ timeEntry, timeEntriesIndex }) => {
+  const { timeEntries, setTimeEntries, timecardDate } = useTimecardContext();
+  const timeEntryExists = !!timeEntry.timeEntryId;
+
+  const handleClickRemoveButton = async (event) => {
+    event.preventDefault();
+    const params = new UrlSearchParamBuilder()
+      .setTenantId('00000000-0000-0000-0000-000000000000')
+      .getUrlSearchParams();
+    const response = await deleteTimeEntry(timeEntry.timeEntryId, params);
+
+    if (response.status === 200) {
+      removeTimecardContextEntry(timeEntries, setTimeEntries, timeEntriesIndex);
+    }
+  };
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+
+    const actualStartDateTime = formatDateTimeISO(
+      dayjs(timecardDate).startOf('day')
+    );
+    const actualEndDateTime = formatDateTimeISO(
+      dayjs(timecardDate).startOf('day').add(1, 'day')
+    );
+
+    const timecardPayload = {
+      ownerId: 1,
+      timePeriodTypeId: timeEntry.timePeriodTypeId,
+      actualStartTime: actualStartDateTime,
+      actualEndTime: actualEndDateTime,
+    };
+
+    const params = new UrlSearchParamBuilder()
+      .setTenantId('00000000-0000-0000-0000-000000000000')
+      .getUrlSearchParams();
+
+    const response = await createTimeEntry(timecardPayload, params);
+
+    if (response?.data?.items?.length > 0) {
+      const newTimeEntries = deepClone(timeEntries);
+      newTimeEntries[timeEntriesIndex] = ContextTimeEntry.createFrom(timeEntry)
+        .setStartTime(midnight)
```

Comment:
shouldn't start and end time come from the successful response? As per TimeEntryId

Reply:
Updated this now